tests/units/RuleRight.php

```diff
@@ -152,4 +152,63 @@ public function testLocalAccount() {
       }
       $this->boolean($found)->isFalse();
    }
+
+   public function testLocalAccountNoRules() {
+      $testuser = [
+         'name'      => 'testuser',
+         'password'  => 'test',
+         'password2' => 'test',
+      ];
+
+      $user = new \User();
+      $users_id = $user->add($testuser);
+      $this->integer($users_id)->isGreaterThan(0);
+
+      // Get rights
+      $pu = \Profile_User::getForUser($users_id, true);
+
+      // User should have a single dynamic right
+      $this->array($pu)->hasSize(1);
+      $right = array_pop($pu);
+      $this->integer($right['is_dynamic'])->isEqualTo(1);
+      $this->integer($right['is_default_profile'])->isEqualTo(1);
+
+      // Log in to force rules right processing
+      $this->login($testuser['name'], $testuser['password'], false);
+
+      // Get rights again, should not have changed
+      $pu = \Profile_User::getForUser($users_id, true);
+      $this->array($pu)->hasSize(1);
+      $right2 = array_pop($pu);
+      $this->array($right2)->isEqualTo($right);
+
+      $_SESSION['glpiactiveprofile']['user'] = 7327;
+      $_SESSION['glpiactiveprofile']['profile'] = 23;
+
```

Comment:
```suggestion
```

Reply:
Does not work without, seems like there is some rights checks inside the DB engine for this case :/

Reply:
Where does thes values comes ?

Reply:
empty_data.php, profiles rights for a super admin:
![image](https://user-images.githubusercontent.com/42734840/108345705-e3980000-71de-11eb-9143-3045340e7265.png)

Reply:
Use `$this->login();` to login with TU_USER which has these rights, instead of forcing value.

Reply:
I've tried it before and it didn't work but it seemed weird since as you said TU_USER have the two rights that I was adding.
I just found out that you also need to be the one that created the user to be allowed to edit his profiles, so adding a second `$this->login();` before the creating the user at line 164 make it works.

Is it normal to have these rights issues ?
Shouldn't rights be managed in an upper layer (front files, controllers, ...) and not intervene in this kind of situations ?

Reply:
> I just found out that you also need to be the one that created the user to be allowed to edit his profiles

Are you sure of that ?

I think it was failing before because, and it is normal, a user cannot add itself a profile that gives him more rights. As in your test, you were connected as the tested user `$this->login($testuser['name'], $testuser['password'], false);`, it is normal that you were not able to add a profile for this user.